src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,320 @@
+import React, { useContext } from 'react'
+import { add, addDays, addMonths, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '..'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./TimeSeriesChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import { ceil } from 'lodash'
+import addHours from 'date-fns/addHours'
+
+export interface TimeSeriesChartProps {
+    title?: string
+    intervalType?: "Week" | "Month" | "SixMonth" | "Day",
+    intervalStart: Date,
+    data: any[] | undefined,
+    expectedDataInterval?: Duration,
+    dataKeys?: string[],
+    chartHasData: boolean,
+    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
+    chartType: "Line" | "Bar" | "Area"
+    options?: LineChartOptions | BarChartOptions | AreaChartOptions
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface LineChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    connectNulls?: boolean,
+    domainMin?: number | "Auto"
+}
+
+export interface BarChartOptions {
+    barColor?: ColorDefinition | ColorDefinition[],
```

Comment:
I wrestled with this a bit because the parallel arrays didn't make me very happy either.  I was trying to walk a balance of not having 3 different interfaces for doing things between 3 different related but different widgets (the DailyDataChart, the SurveyAnswerChart, and the TimeSeriesChart).  I was trying to avoid having the DailyData and SurveyAnswer have "custom" interfaces that they each "adapted" into something for the TimeSeriesChart because that seemed more confusing.  In particular I really wanted them to be able to share these base FooChartOptions classes.  And in doing so, I was trying to not break the current interface of the DailyDataChart, so that any current users could keep on working, and I wouldn't have to change View Builder.  

Honestly, I feel strongly that changing the dataKeys for the TimeSeriesChart into objects instead of the current string array makes it clearer.  I think that having two separate places where you can define the colors is harder to discover and understand as an end user, and the fact that we'd have to convey the hierarchy of where we look for and prioritize those colors is a situation I'd prefer to avoid.

Here's two suggestions:

1) Move the `dataKeys` property off of the `TimeSeriesChartProps` and into each `FooChartOptions`.
```
export interface LineChartOptions {
    lineColor?: ColorDefinition | ColorDefinition[],
    dataKeys?: undefined | string[],
    connectNulls?: boolean,
    domainMin?: number | "Auto"
}

export interface BarChartOptions {
    barColor?: ColorDefinition | ColorDefinition[],
    dataKeys?: undefined | string[],
    thresholds?: BarChartThreshold[]
}
```
This at least *colocates* the parallel arrays so they're next to each other.  

2) Similar solution, but more like your suggestion, again removing the `dataKeys` property from `TimeSeriesChart`:
```
export interface LineChartOptions {
    lineColor?: ColorDefinition,
    seriesDefinition? : undefined | { dataKey: string, lineColor: ColorDefinition }[]
    connectNulls?: boolean,
    domainMin?: number | "Auto"
}

export interface BarChartOptions {
    barColor?: ColorDefinition,
    seriesDefinition? : undefined | { dataKey: string, barColor: ColorDefinition }[]
    thresholds?: BarChartThreshold[]
}
```
This has the issue of having to "prioritize" things, but at least in this case the properties are much closer together and maybe its more obvious?

Reply:
Hmm...yeah neither option here feels right to me.

1 as you point out is still parallel arrays

2 is really burying a key required top-level property into sub-sub object so our chart would be like ```<TimeSeriesChart options={{seriesDefinition:{dataKey:"Steps", barColor:"blue"}}}/>```
which feels rough.

I would certainly not break the DailyDataChart interface/props, but I think it'd be ok to have a distinct DailyDataChartLineOptions, DailyDataChartBarChartOptions, etc. interfaces so we have freedom here...just changing types is not enough of a breaking change that I would feel like we'd need to cut a new major.  Or we can name these new options "TimeSeriesChartOptions"

So I think we do that^ and keep those types the same.  And then we can split our "options" from "series" such that we have:

```
export interface TimeSeriesChartProps {
    title?: string
    intervalType?: "Week" | "Month" | "6Month" | "Day",
    intervalStart: Date,
    data: any[] | undefined,
    expectedDataInterval?: Duration,
    series: ChartSeries[]
    chartHasData: boolean,
    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
    chartType: "Line" | "Bar" | "Area"
    options?: TimeSeriesLineChartOptions | TimeSeriesBarChartOptions
    innerRef?: React.Ref<HTMLDivElement>
}

export interface ChartSeries {
    dataKey: string
    color: ColorDefinition
} 

export interface AreaChartSeries extends ChartSeries {
    areaColor: ColorDefinition // not sure how many use cases for multi-series area charts we have (or area charts in general), so could go either way on this
} 

export interface TimeSeriesLineChartOptions {
    connectNulls?: boolean,
    domainMin?: number | "Auto"
}

export interface TimeSeriesBarChartOptions {
    thresholds?: BarChartThreshold[]
}

export interface TimeSeriesBarChartThreshold {
    value: number
    referenceLineColor?: ColorDefinition
    overThresholdBarColor?: ColorDefinition
}
```

Something like that?  And now that I look at that ^ we could consider dropping "options" entirely.  ```domainMin ``` probably makes sense for all charts I think...not sure why it was just added to bars.  And sounds like we're adding thresholds to line charts as well...not sure if that is coming here or in @reneefromhold 's PR.  The only kinda oddball one feels like "connectNulls" which i'd be fine just putting on TimeSeriesChartProps even if it doesn't make sense for bar/area charts right now.

Reply:
I've implemented this with the slight adjustment that instead of calling it `TimeSeriesLineChartOptions` I called them `MultipleSeriesLineChartOptions` because then the `SurveyAnswerChart` and `TimeSeriesChart` both share a similar interface because they both support showing multiple series of data.  The `DailyDataChart` is the only one that converts from its own property styles to the interface that `TimeSeriesChart` wants.  I've adjusted all the stories as necessary.

I'd talked to Renee and asked if she would add thresholds to line charts in her PR as this PR was large enough as it was, and adding it in her PR where it would be used the first time seemed appropriate.